File: src/main/java/com/regionrental/commands/InfoCommand.java
package com.regionrental.commands;

import com.regionrental.RegionRental;
import com.regionrental.managers.Rental;
import com.regionrental.util.WorldRegionParser;
import org.bukkit.ChatColor;
import org.bukkit.World;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

public class InfoCommand implements CommandExecutor {
    
    private final RegionRental plugin;
    
    public InfoCommand(RegionRental plugin) {
        this.plugin = plugin;
    }
    
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!sender.hasPermission("regionrental.info")) {
            sender.sendMessage(plugin.getConfigManager().getMessage("no-permission"));
            return true;
        }
        
        if (args.length < 1) {
            sender.sendMessage(ChatColor.RED + "Usage: /rrinfo <world:region>");
            sender.sendMessage(ChatColor.YELLOW + "Example: /rrinfo world:shop1");
            return true;
        }

        // Parse region argument with world inference
        WorldRegionParser.ParsedRegion parsed = WorldRegionParser.parse(args[0], sender);
        if (parsed == null) {
            sender.sendMessage(ChatColor.RED + "Invalid format! Console must use world:region format (e.g., world:shop1)");
            return true;
        }

        World world = parsed.getWorld();
        String regionName = parsed.regionName;

        // Check if region exists
        if (!plugin.getWorldGuardManager().regionExists(regionName, world)) {
            sender.sendMessage(plugin.getConfigManager().getMessage("region-not-found",
                "{region}", parsed.getCompositeKey()));
            return true;
        }

        Rental rental = plugin.getRentalManager().getRental(regionName, world);

        sender.sendMessage(ChatColor.GOLD + "=== Region: " + parsed.getCompositeKey() + " ===");

        if (rental == null) {
            sender.sendMessage(ChatColor.GREEN + "Status: " + ChatColor.WHITE + "Available");
            sender.sendMessage(ChatColor.YELLOW + "Price: " + ChatColor.WHITE +
                String.format(plugin.getConfigManager().getCurrencyFormat(),
                    plugin.getConfigManager().getPriceForRegion(regionName, world)));
            sender.sendMessage(ChatColor.YELLOW + "Duration: " + ChatColor.WHITE +
                plugin.getConfigManager().getDurationForRegion(regionName, world) + " days");
        } else {
            sender.sendMessage(ChatColor.RED + "Status: " + ChatColor.WHITE + "Rented");
            sender.sendMessage(ChatColor.YELLOW + "Owner: " + ChatColor.WHITE + rental.getPlayerName());
            sender.sendMessage(ChatColor.YELLOW + "Expires: " + ChatColor.WHITE + rental.getFormattedEndDate());
            sender.sendMessage(ChatColor.YELLOW + "Time Remaining: " + ChatColor.WHITE + 
                rental.getDaysRemaining() + " days, " + 
                (rental.getHoursRemaining() % 24) + " hours");
            sender.sendMessage(ChatColor.YELLOW + "Extensions Used: " + ChatColor.WHITE + 
                rental.getExtensionCount() + "/" + plugin.getConfigManager().getMaxExtensions());
            sender.sendMessage(ChatColor.YELLOW + "Total Paid: " + ChatColor.WHITE + 
                String.format(plugin.getConfigManager().getCurrencyFormat(), rental.getTotalPaid()));
        }
        
        return true;
    }
}

File: src/main/kotlin/com/zonerental/commands/InfoCommand.kt
package com.zonerental.commands

import com.zonerental.ZoneRental
import com.zonerental.extensions.sendMiniMessage
import com.zonerental.util.WorldRegionParser
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class InfoCommand(private val plugin: ZoneRental) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (!sender.hasPermission("zonerental.info")) {
            sender.sendMessage(plugin.configManager.getMessage("no-permission"))
            return true
        }

        if (args.isEmpty()) {
            sender.sendMiniMessage("<red>Usage: /zrinfo <world:region>")
            sender.sendMiniMessage("<yellow>Example: /zrinfo world:shop1")
            return true
        }

        // Parse region argument with world inference
        val parsed = WorldRegionParser.parse(args[0], sender) ?: run {
            sender.sendMiniMessage("<red>Invalid format! Console must use world:region format (e.g., world:shop1)")
            return true
        }

        val world = parsed.getWorld() ?: run {
            sender.sendMiniMessage("<red>World not found!")
            return true
        }
        val regionName = parsed.regionName

        // Check if region exists
        if (!plugin.worldGuardManager.regionExists(regionName, world)) {
            sender.sendMessage(plugin.configManager.getMessage("region-not-found", "{region}", parsed.getCompositeKey()))
            return true
        }

        val rental = plugin.rentalManager.getRental(regionName, world)

        sender.sendMiniMessage("<gold>=== Region: ${parsed.getCompositeKey()} ===")

        if (rental == null) {
            // Region is available
            val price = String.format(plugin.configManager.currencyFormat,
                plugin.configManager.getPriceForRegion(regionName, world))
            val duration = plugin.configManager.getDurationForRegion(regionName, world)

            sender.sendMiniMessage("<green>Status: <white>Available")
            sender.sendMiniMessage("<yellow>Price: <white>$price")
            sender.sendMiniMessage("<yellow>Duration: <white>$duration days")
        } else {
            // Region is rented
            val hoursRemaining = rental.hoursRemaining % 24
            val totalPaid = String.format(plugin.configManager.currencyFormat, rental.totalPaid)

            sender.sendMiniMessage("<red>Status: <white>Rented")
            sender.sendMiniMessage("<yellow>Owner: <white>${rental.playerName}")
            sender.sendMiniMessage("<yellow>Expires: <white>${rental.formattedEndDate}")
            sender.sendMiniMessage("<yellow>Time Remaining: <white>${rental.daysRemaining} days, $hoursRemaining hours")
            sender.sendMiniMessage("<yellow>Extensions Used: <white>${rental.extensionCount}/${plugin.configManager.maxExtensions}")
            sender.sendMiniMessage("<yellow>Total Paid: <white>$totalPaid")
        }

        return true
    }
}
